### PR TITLE
[WIP] pdf conversion job

### PR DESCRIPTION
**Pull request by kreek**

### CODE REVIEW

File: spec/jobs/document_uploads/pdf_conversion_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe DocumentUploads::PdfConversionJob do
+  describe '#perform' do
+    before(:all) do
+      @file_path = Rails.root.join('spec/fixtures/files/pdf_conversion.jpg')
```
Comment (omgitsbillryan) [inline spec/jobs/document_uploads/pdf_conversion_job_spec.rb]: I think it's supposed to be 
```
@file_path = Rails.root.join('spec', 'fixtures', 'files', 'pdf_conversion.jpg')
```

in the unlikely even that this spec is run on an OS where `/` is not the default separator

File: spec/jobs/document_uploads/pdf_conversion_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      @file_path_converted = Rails.root.join('spec/fixtures/files/pdf_conversion.jpg.pdf')
+      MiniMagick::Tool::Convert.new do |convert|
```
Comment (omgitsbillryan) [inline spec/jobs/document_uploads/pdf_conversion_job_spec.rb]: This is creating the .jpg right?  Can this be abstracted into a helper for readability?

[SO Post](http://stackoverflow.com/a/8127728)

**Comment by markolson** [comment]
I suspect we're going to need  `pandoc` in order to handle .txt and .doc files along with imagemagick for the images.

**Comment by kreek** [comment]
@markolson cc @jcmeloni-usds @J-Quag are word docs a requirement for the first iteration of document uploads? Devops spidey-sense understandably being triggered by allowing files with macros. We'll have a virus scan that _should_ block those files but it's safer to do only lossless conversion, as with image to PDF, than word to PDF which requires Office or OSS clone (LibreOffice) to reliably convert.

**Comment by J-Quag** [comment]
So it is an interesting question. In EVSS we actually can’t do anything with Word, Excel, PPT, etc due apparently to licensing issues.
   If there is a macro concern, let’s make sure we don’t have any vulnerabilities there. If Word is a problem, it hasn’t been something we’ve allowed, but would like to add it if its not a risk.

**Comment by markolson** [comment]
HCA backend allowable types look like Word, PDF, RTF and JPG. Not sure what EDU supports, but we can probably just say TXT/RTF, Image and PDF and maybe have a link to print-to-pdf instructions for Word?

**Comment by kreek** [comment]
New [PR](https://github.com/department-of-veterans-affairs/vets-api/pull/1011) for this (as workflow task) against master instead of topic branch

**Pull request closed without merging**